theseus/core/cost_function.py

```diff
@@ -240,6 +241,17 @@ def _compute_autograd_jacobian_vmap(
         aux_tensors: Tuple[torch.Tensor, ...],
         jac_fn: Callable,
     ) -> Tuple[torch.Tensor, ...]:
+        def _expand_all(
+            tensors: Tuple[torch.Tensor, ...], batch_size: int
+        ) -> Tuple[torch.Tensor, ...]:
+            return tuple(
+                t if t.shape[0] == batch_size else t.expand((batch_size,) + t.shape[1:])
+                for t in tensors
+            )
+
+        batch_size = max(t.shape[0] for t in chain(optim_tensors, aux_tensors))
+        optim_tensors = _expand_all(optim_tensors, batch_size)
+        aux_tensors = _expand_all(aux_tensors, batch_size)
```

Comment:
I assume this is more related with vectorization. Do we need to guarantee there are no multiple `batch_size!=1`?